File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Java/payment/Coupon.java
package payment;

public class Coupon {
    private String code;
    private double discountPercentage;
    private boolean isActive;

    Coupon(String code, double discountPercentage, boolean isActive) {
        this.code = code;
        this.discountPercentage = discountPercentage;
        this.isActive = isActive;
    }

    public String getCode() {
        return code;
    }

    public double getDiscountPercentage() {
        return discountPercentage;
    }

    public boolean getIsActive() {
        return isActive;
    }
}

File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Cpp/payment/src/Coupon.cpp
#include "../include/Coupon.h"

Coupon::Coupon() 
    : code(""), discountPercentage(0.0), isActive(false) {}

Coupon::Coupon(const std::string& code, double discountPercentage, bool isActive)
    : code(code), discountPercentage(discountPercentage), isActive(isActive) {}

const std::string& Coupon::getCode() const {
    return code;
}

double Coupon::getDiscountPercentage() const {
    return discountPercentage;
}

bool Coupon::getIsActive() const {
    return isActive;
}